docs/developer-setup.md

```diff
@@ -0,0 +1,234 @@
+# Nextcloud Talk development setup
+
+The following guide is intended to setup an environment for Nextcloud Talk development. It is _not_ intended to be used in production at any time.
+
+## Preface
+
+Nextcloud Talk delivers private audio/video conferencing and text chat through browser and mobile apps. It consists of several components, some of which are optional:
+
+* **Nextcloud Server**: The core component of Nextcloud that offers file storage capabilities and the framework for building/running apps.
+
+* **Nextcloud Talk Server**: The Nextcloud app that is the core of Nextcloud Talk. It consists of the backend (PHP) and a frontend (Vue.js) to use Nextcloud Talk directly in the web browser. **Important**: Due to legacy reasons, the app id of "talk" is "spreed"!
+
+* **High performance backend**: _(optional)_ The high performance backend is an optional component that consists of a signaling server and a WebRTC gateway. It ensures that calls with more than 4 participants work smoothly. It is also required for features like typing indicators or federated calling to work properly.
+
+* **Nextcloud Talk Recording Server**: _(optional)_ The Nextcloud Talk recording server is required when calls should be recorded on the server side. It takes care of the recording itself and uploading the result to Nextcloud.
+
+* **Nextcloud Talk Android**: _(optional)_ Nextcloud Talk Android is the recommended way of using Nextcloud Talk on Android devices. Mostly written in Kotlin nowadays, it still contains some Java code.
+
+* **Nextcloud Talk iOS**: _(optional)_ The recommended way of using Nextcloud Talk on your iOS devices. Major parts are still written in Objective-C while the transition to Swift is an ongoing effort.
+
+## Prerequirements
+
+There are many ways of running Nextcloud for development purposes. In the following guide we are using a docker based installation of Nextcloud. The following components are required:
+
+* A working docker installation (see https://docs.docker.com/engine/install/)
+* Node.js >= 20 and NPM >= 10 (optionally, see _Prepare Nextcoud Talk for development_ below for an alternative)
+* PHP Composer (optionally, see _Prepare Nextcoud Talk for development_ below for an alternative)
+* Git
+
+## Install the development environment
+
+Clone `nextcloud-docker-dev` in a directory of your choice:
+
+```
+cd /path/to/directory
+git clone https://github.com/juliushaertl/nextcloud-docker-dev
+cd nextcloud-docker-dev
+```
+
+`bootstrap.sh` is used to setup the enviroment, it supports multiple parameters that may depend on your development focus:
+
+```
+boottrap.sh [--full-clone|--clone-no-blobs] [--clone-all-apps-filtered] [--] APPS
```

Comment:
```suggestion
bootstrap.sh [--full-clone|--clone-no-blobs] [--clone-all-apps-filtered] [--] APPS
```

Reply:
Good catch, will also fix it upstream: https://github.com/juliushaertl/nextcloud-docker-dev/blob/f410d902f3f89f527b9e95af38ed82b0c4c3130a/bootstrap.sh#L14